language/golang/build-images.md

```diff
@@ -0,0 +1,388 @@
+
+```dockerfile
+WORKDIR /app
+```
+
+Usually the very first thing you do once you’ve downloaded a project written in Go is to install the modules necessary to compile it.
+
+Before we can run `go mod download`, we need to get our `go.mod` and `go.sum` files into our images. We use the `COPY` command to do this. 
+
+In its simplest form, the `COPY` command takes two parameters. The first parameter tells Docker what file you would like to copy into the image. The second parameter tells Docker where you want that file to be copied to. 
+
+We’ll copy the `go.mod` and `go.sum` file into our working directory `/app` which, owing to our use of `WORKDIR`, is the current directory (`.`)
+
+```dockerfile
+COPY go.mod .
+COPY go.sum .
+```
+
+Now that we have the module files inside the Docker image that we are building, we can use the `RUN` command to execute the command `go mod download` there as well. This works exactly the same as if we were running `go` locally on our machine, but this time these Go modules will be installed into the a directory inside our image.
+
+```dockerfile
+RUN go mod download
+```
+
+At this point, we have an image that is based on Go environment version 1.16 (or a later minor version, since we had specified `1.16` as our tag in the `FROM` command) and we have installed our dependencies. 
+
+The next thing we need to do is to add our source code into the image. We’ll use the `COPY` command just like we did with our module files before.
+
+```dockerfile
+COPY *.go .
+```
+
+This `COPY` command uses a wildcard to copy all files with `.go` extension located in the current directory into the image. 
+
+Now, we would like to compile our application. To that end, we use the familiar `RUN` command.
+
+```dockerfile
+RUN go build -o /docker-gs-ping
+```
+
+This should be familiar. The result of that command will be a static application binary named `docker-gs-ping` and located in the root of the filesystem of the image that we are building. We could have put the binary into any other place we desire inside that image, there is no special significance for placing it in the root.
+
+Now, all that is left to do is to tell Docker what command to execute when our image is used to start a container. 
+
+We do this with the `CMD` command.
+
+```dockerfile
+CMD [ "/docker-gs-ping" ]
+```
+
+Here's the complete `Dockerfile`. The file in the repository that you cloned may also contain comments. They always begin with a `#` symbol and make no difference to Docker. The comments are there for the convenience of humans tasked to maintain the `Dockerfile`.
+
+```dockerfile
+FROM golang:1.16-alpine
+
+WORKDIR /app
+
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
+
+COPY *.go .
+
+RUN go build -o /docker-gs-ping
+
+EXPOSE 8080
+
+CMD [ "/docker-gs-ping" ]
+```
+
+You can also find this `Dockerfile` for this part in [go-docker](https://github.com/olliefr/go-docker) repo.
```

Comment:
The link is wrong, should it be https://github.com/olliefr/docker-gs-ping ?

Reply:
That sentence shouldn't be there at all. Removed!